### PR TITLE
NIFI-6740: Add configuration options to specify NiFi/Bootstrap communication ports

### PR DESCRIPTION
**Pull request by stevedlawrence**
#### Description of PR

The NiFi and NiFi Bootstrap processes both bind to random ephemeral
ports to allow for inter-process communication (e.g. shutdown, port,
ping, etc.). However, the randomness of these ephemeral ports can pose
challenges for some security policies (e.g. iptables, firewall).

This adds two configuration options, nifi.bootstrap.listen.port and
nifi.commandcontrol.listen.port, that allow an administrator to define
which ports the two processes should bind to for this communication,
making it easier to define security policies. Options default to zero to
maintain the current ephemeral port behavior.

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [X Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [X] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java [hunk 1]
```diff
@@ -81,6 +81,7 @@
     public static final String PROCESSOR_SCHEDULING_TIMEOUT = "nifi.processor.scheduling.timeout";
     public static final String BACKPRESSURE_COUNT = "nifi.queue.backpressure.count";
     public static final String BACKPRESSURE_SIZE = "nifi.queue.backpressure.size";
+    public static final String COMMANDCONTROL_LISTEN_PORT = "nifi.commandcontrol.listen.port";
```
Comment (mattyb149) [inline nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java]: Command & Control already has a different meaning in the NiFi ecosystem context: https://cwiki.apache.org/confluence/display/MINIFI/C2+Design

In the code we already refer to it as a "bootstrap listener", so maybe `nifi.bootstrap.port` and `nifi.bootstrap.listener.port` are better names for the properties?

Comment (stevedlawrence) [reply nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java]: Thanks for taking the time to review this!

Makes sense. I've renamed the property to ``nifi.bootstrap.port``, and rebased to master.

File: nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java [hunk 2]
```diff
@@ -81,6 +81,7 @@
     public static final String PROCESSOR_SCHEDULING_TIMEOUT = "nifi.processor.scheduling.timeout";
     public static final String BACKPRESSURE_COUNT = "nifi.queue.backpressure.count";
     public static final String BACKPRESSURE_SIZE = "nifi.queue.backpressure.size";
+    public static final String BOOTSTRAP_PORT = "nifi.bootstrap.port";
```
Comment (MuazmaZ) [inline nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java]: Could you rename this to nifi listener bootstrap port as well to keep in sync?

Comment (stevedlawrence) [reply nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java]: Thanks for reviewing. THis is been fixed, squashed, and rebased to the latest master.

**Pull request closed without merging**